.gitmodules

```diff
@@ -31,3 +31,6 @@
 [submodule "dependencies/benchmark"]
 	path = dependencies/benchmark
 	url = https://github.com/google/benchmark.git
+[submodule "dependencies/cxxopts"]
```

Comment:
The problem with having cxxopts as submodule (not a bad idea in itself) is that when git is not available, then we cannot build the tools. Because we are importing literally just a single file (cxxopts.hpp), I think it might be simpler to just copy the file (cxxopts.hpp) in the project. Here is my proposal: just drop a copy of cxxopts.hpp in tools. This is not ideal in that we will want to use cxxopts.hpp elsewhere (e.g., in benchmarking) but it might be a good start.

Reply:
I actually prefer having the submodule where possible, because I've seen so many projects ... but I see your point. I could live with a rule of "tools and library must build without git." Where should it go, though?

If we drop cxxopts.hpp into the project somewhere, make certain to put a comment at the very top stating the URL and project it comes from, so we can update it. And maybe a massive comment to make it super clear:

```c++
/**
 *
 * YOU.
 *
 * THAT'S RIGHT, YOU.
 *
 * DO NOT MODIFY THIS ON PAIN OF DEATH.
 *
 * PULL NEW COPIES FROM THE SOURCE.
 *
 * WE'LL BE WATCHING YOU.
 *
 */
```

Reply:
I think it is was ok to have it as a submodule, and maybe we can leave it as a submodule, but we need to pass CI tests.

Reply:
Thinking about it, I now agree that leaving things as they are (with a submodule) is fine.

 But we should be consistent and do the same with toni_ronnko_dirent.h.